WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -1221,27 +1207,36 @@ private void savePostAndFinish() {
                 }
             }
 
-            if (PostStatus.fromPost(mPost) != PostStatus.PUBLISHED && isPublishable) {
-                publishPost();
+            if (PostStatus.fromPost(mPost) != PostStatus.PUBLISHED && isPublishable && NetworkUtils.isNetworkAvailable(this)) {
+                if (!hasFailedMedia()) {
+                    savePostOnlineAsyncAndFinish(isFirstTimePublish);
+                }
             } else {
-                savePostLocally();
+                savePostLocallyAsyncFinish();
             }
+        } else {
+            // discard post if new & empty
+            if (!isPublishable && isNewPost()) {
+                mDispatcher.dispatch(PostActionBuilder.newRemovePostAction(mPost));
+            }
+            finish();
         }
+    }
 
-        // discard post if new & empty
-        if (isNewPost() && !isPublishable) {
-            mDispatcher.dispatch(PostActionBuilder.newRemovePostAction(mPost));
-        }
+    private boolean updatePostObject() {
```

Comment:
Hey @0nko ! This version of `updatePostObject` is quite different to the `updatePostObject(boolean)` one.

* It calculates and returns a boolean but doesn't use it internally
* The other version of updatePostObject doesn't return a value at all

Because of these, I would suggest splitting this into a `boolean isFirstTimePublish()` and a `void updatePostObject()` for clearer functionality.

Also, it seems that it catches the `IllegalEditorStateException`, thus allowing its caller to continue normally, without a user notification. Not sure at this point but, can you confirm that this is indeed the intended behaviour here? The other version of `updatePostObject` does throw it. Thanks!